docs/adr/001-project-locks.md

```diff
@@ -0,0 +1,160 @@
+
```

Comment:
I'd follow a process like adr-tools on this

https://github.com/npryce/adr-tools/tree/master/doc/adr

The first adr is meta which is to document the adrs and ensure they follow a pattern and ensure immutability

Reply:
agree, might be good to introduce some meta ADR doc first :)

Reply:
good call outs, I'll rename this to 002 and make a doc on the ADR process as 001

Reply:
Great!

Can we follow the same format of 0001 too?

The more consistent with other adr standards, the less creative we'll have to be

Reply:
@chenrui333 @nitrocode I added the 0001 doc here and renamed the project-locks doc